File: TriCoachAI/Models/AdaptationEvent.swift
import Foundation

struct AdaptationEvent: Codable, Equatable, Identifiable {
    var id: UUID = UUID()
    var planId: UUID
    var triggeredBy: AdaptationTrigger
    var actionTaken: String
    var deltaLoadPercent: Double?
    var createdAt: Date = .now
}

File: TriCoachAndroid/app/src/main/java/com/tricoach/android/models/AdaptationEvent.kt
package com.tricoach.android.models

import kotlinx.serialization.Serializable

@Serializable
data class AdaptationEvent(
    val id: String,
    val planId: String,
    val triggeredBy: AdaptationTrigger,
    val actionTaken: String,
    val deltaLoadPercent: Double? = null,
    val createdAt: String = "",
)
